tcl/org.testeditor.tcl.dsl/src/org/testeditor/tcl/dsl/jvmmodel/TclJvmModelInferrer.xtend

```diff
@@ -193,6 +200,18 @@ class TclJvmModelInferrer extends AbstractModelInferrer {
 			annotations += annotationRef('org.junit.Test') // make sure that junit is in the classpath of the workspace containing the dsl
 			body = [element.generateMethodBody(trace(element, true))]
 		]
+
```

Comment:
It's pretty much the same. The difference is that the generation for the test first has to iterate over all specification steps and then calls `generateContext` for each context while the macro can directly do that.